### ISSUE TITLE
Prow does not have env var PULL_BASE_REF set for periodic jobs

### ISSUE THREAD

**Issue by Fredy-Z** [knative/test-infra#780]
This issue is used to track one potential issue for Prow.

Background:
We need to use `PULL_BASE_REF` to get the actual branch name for the repo running as Prow jobs, but  currently Prow only set it for non-periodic jobs, see https://github.com/kubernetes/test-infra/blob/master/prow/jobs.md. 
We are temporarily adding it in our periodic job configs, and checking with Kubernetes test-infra if we can add it in all Prow periodic jobs. If we can, we will create a PR for Prow and delete our own logic.

**Comment by adrcunha** [comment]
@Fredy-Z can you please file an issue in kubernetes/test-infra and link it from here?

**Comment by Fredy-Z** [comment]
I checked with Cole, he said since periodic job is not necessarily meant to run against specific repo, it can be any kind of job, that's why `PULL_BASE_REF` is not required in its environment variables.
And also since we will specify the branch name for our periodic jobs, that's one thing we've already known, adding it as the environment variable for prow jobs will be confusing and kind of misleading.

Based on these facts, I think our current solution is probably already the best one.

However, when I checked our current prow config, I found the `branch name` was missed in the CI jobs for releases, see https://github.com/knative/test-infra/blob/9ccec33ae7eb8ede1bdffeb3dc50b0b6ec7b6c88/ci/prow/config.yaml#L1864, now all CI jobs are actually running against `master` branch. The broken config started from Chao's PR to replace `bootstrap` with `podutil`. I will create an issue for it.
